### PR TITLE
Adding a frame collection parser and heartbeat sequence container

### PR DESCRIPTION
**Pull request by matthiasrichter**
- adding a generic reverse parser for sequences of blocks composed
  of header - payload - trailer with variable payload length
- adding container class for multiple sequences of heartbeat frames
  with row and column access iterators

Further comments:
passing the access function to the iterator did not work because
the typedef for the access function is without the capture, so there
is no matching conversion.

Solution would be to use std::function but that's probably slow and
the function is called often. Can be checked later.

### CODE REVIEW

File: DataFormats/Headers/include/Headers/HeartbeatFrame.h [hunk 1]
```diff
@@ -74,24 +83,26 @@ struct HeartbeatTrailer
...
-	// bit 55: =1 HBa/0 HBr received
-	uint16_t hbAccept:1;
-	// bit 56 to 59: trailer length
-	uint16_t trailerLength:4;
-	// bit 60 to 63: block type (=5 HBF Trailer)
-	uint8_t blockType:4;
+        // bit 0 to 31: data length in words
+        uint32_t dataLength;
+        // bit 32 to 52: detector specific status words
+        uint32_t status:21;
```
Comment (sawenzel) [inline DataFormats/Headers/include/Headers/HeartbeatFrame.h]: why these changes?

File: DataFormats/Headers/CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +25,7 @@ O2_GENERATE_LIBRARY()
 set(TEST_SRCS
   test/testDataHeader.cxx
   test/testTimeStamp.cxx
+  test/test_HeartbeatFrame.cxx
```
Comment (sawenzel) [inline DataFormats/Headers/CMakeLists.txt]: the name does not look  consistent with the other tests

File: DataFormats/Headers/include/Headers/HeartbeatFrame.h (hunk 1 above)
Comment (sawenzel) [reply DataFormats/Headers/include/Headers/HeartbeatFrame.h]: @matthiasrichter ...

Comment (matthiasrichter) [reply DataFormats/Headers/include/Headers/HeartbeatFrame.h]: tabs has been replaced with blanks

File: DataFormats/Headers/CMakeLists.txt (hunk 2 above)
Comment (matthiasrichter) [reply DataFormats/Headers/CMakeLists.txt]: a naming scheme will only be imposed in #546 and uses the `test_` prefix. But then the prefix is automatically added. In that respect the old names are not consistent. I expect this to be changed anyhow when #546 is merged.

File: DataFormats/Headers/include/Headers/HeartbeatFrame.h (hunk 1 above)
Comment (sawenzel) [reply DataFormats/Headers/include/Headers/HeartbeatFrame.h]: the old and new indentations are not consistent with clang-format. But, given that we will fix this globally at some point I will merge this now.
